pkg/server/server.go

```diff
@@ -275,7 +275,6 @@ func (s *Server) HTTPHandler() http.Handler {
 			tenancyRulesSourcePath      = prometheusTenancyProxyEndpoint + "/api/v1/rules"
 			tenancyTargetAPIPath        = prometheusTenancyProxyEndpoint + "/api/"
 
-			prometheusProxy            = proxy.NewProxy(s.PrometheusProxyConfig)
```

Comment:
If remove `prometheusProxy`, are we also able to remove `PrometheusProxyConfig` from main.go and server.go? I suspect we'd also be able to remove `openshiftPrometheusHost` from main.go and the `k8s-mode-off-cluster-prometheus` flag.

Reply:
After thinking more about it and discussing with @kyoto, we may still need the Prometheus proxy eventually because we want to expose Prometheus targets in the admin console. Thanos doesn't expose this information right now but if it does at some point, we could then delete the Prometheus proxy bits.

Reply:
Update: after more research, it appears that Thanos envisions adding a new endpoint to federate Prometheus `/api/v1/targets` endpoints (https://github.com/thanos-io/thanos/issues/1375). I've updated the PR to remove the Prometheus bits.